heudiconv/dicoms.py

```diff
@@ -402,68 +404,54 @@ def embed_nifti(dcmfiles, niftifile, infofile, bids_info, min_meta):
+      level of the dict structure, no recursive fancy updates) what is obtained
+      from nifti and dicoms
 
     """
     # imports for nipype
     import nibabel as nb
     import os.path as op
     import json
     import re
+    from heudiconv.utils import save_json
+
+    from heudiconv.external.dcmstack import ds
+    stack = ds.parse_and_stack(dcmfiles, force=True).values()
+    if len(stack) > 1:
+        raise ValueError('Found multiple series')
+    # may be odict now - iter to be safe
+    stack = next(iter(stack))
+
+    if not op.exists(niftifile):
+        raise NotImplementedError(
+            "%s does not exist. "
+            "We are not producing new nifti files here any longer. "
+            "Use dcm2niix directly or .convert.nipype_convert helper ."
+            % niftifile
+        )
 
-    if not min_meta:
-        from heudiconv.external.dcmstack import ds
-        stack = ds.parse_and_stack(dcmfiles, force=True).values()
-        if len(stack) > 1:
-            raise ValueError('Found multiple series')
-        # may be odict now - iter to be safe
-        stack = next(iter(stack))
-
-        # Create the nifti image using the data array
-        if not op.exists(niftifile):
-            nifti_image = stack.to_nifti(embed_meta=True)
-            nifti_image.to_filename(niftifile)
-            return ds.NiftiWrapper(nifti_image).meta_ext.to_json()
-
-        orig_nii = nb.load(niftifile)
-        aff = orig_nii.affine
-        ornt = nb.orientations.io_orientation(aff)
-        axcodes = nb.orientations.ornt2axcodes(ornt)
-        new_nii = stack.to_nifti(voxel_order=''.join(axcodes), embed_meta=True)
-        meta = ds.NiftiWrapper(new_nii).meta_ext.to_json()
-
-    meta_info = None if min_meta else json.loads(meta)
+    orig_nii = nb.load(niftifile)
+    aff = orig_nii.affine
+    ornt = nb.orientations.io_orientation(aff)
+    axcodes = nb.orientations.ornt2axcodes(ornt)
+    new_nii = stack.to_nifti(voxel_order=''.join(axcodes), embed_meta=True)
+    meta_info = ds.NiftiWrapper(new_nii).meta_ext.to_json()
+    meta_info = json.loads(meta_info)
 
     if bids_info:
+        # make nice with python 3 - same behavior?
+        meta_info = meta_info.copy()
+        meta_info.update(bids_info)
 
-        if min_meta:
-            meta_info = bids_info
-        else:
-            # make nice with python 3 - same behavior?
-            meta_info = meta_info.copy()
-            meta_info.update(bids_info)
-            # meta_info = dict(meta_info.items() + bids_info.items())
-        try:
-            meta_info['TaskName'] = re.search(
-                r'(?<=_task-)\w+', op.basename(infofile)
-            ).group(0).split('_')[0]
-        except AttributeError:
-            pass
     # write to outfile
-    with open(infofile, 'wt') as fp:
-        json.dump(meta_info, fp, indent=3, sort_keys=True)
-
-    return niftifile, infofile
+    save_json(infofile, meta_info)
```

Comment:
should we keep the same indentation here?

Reply:
I was on an edge for this one, and decided that we better do not delay fixing this inconsistency (which we should have done long ago).  Especially now since in case of min_meta, file would be saved with our "standard" indentation (of 4), thus differ from this one.  So, I decided to stay consistent and just use `save_json` with defaults...
but now you made me look at how we use `save_json` and it seems that we prefer `indent=2` in many places:
```
$> git grep -A20 'save_json(' | grep -e 'save_json' -e 'indent'
heudiconv/bids.py:        save_json(descriptor,
heudiconv/bids.py:        save_json(task_file, fields, indent=2, sort_keys=True, pretty=True)
heudiconv/bids.py:        save_json(jsonfile, json_, indent=2)
heudiconv/bids.py:            save_json(json_phasediffname, json_, indent=2)
heudiconv/bids.py:            save_json(participants_json,
heudiconv/bids.py-                indent=2)
heudiconv/bids.py:            save_json(scans_json,
heudiconv/bids.py-                indent=2)
heudiconv/convert.py:        save_json(filegroup_file, filegroup)
heudiconv/convert.py:        save_json(infofile, meta_info)
heudiconv/dicoms.py:    save_json(infofile, meta_info)
heudiconv/tests/test_utils.py:    save_json(valid_json_file, vcontent)
heudiconv/utils.py:def save_json(filename, data, indent=4, sort_keys=True, pretty=False):
heudiconv/utils.py-    indent : int, optional
heudiconv/utils.py-                data, sort_keys=sort_keys, indent=indent)
```

As a separate PR I will propose making default indent to be `2` and thus removing any "ad-hoc" indent use